projects/sonar/src/app/app.component.ts

```diff
@@ -14,20 +14,64 @@
  * You should have received a copy of the GNU Affero General Public License
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
-import { Component } from '@angular/core';
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import {
+  LangChangeEvent,
+  TranslateService as NgxTranslateService,
+} from '@ngx-translate/core';
 import { TranslateService } from '@rero/ng-core';
+import cookie from 'cookie';
+import { Subscription } from 'rxjs';
 
 @Component({
   selector: 'sonar-root',
-  templateUrl: './app.component.html'
+  templateUrl: './app.component.html',
 })
-export class AppComponent {
+export class AppComponent implements OnDestroy, OnInit {
+  // Change language subscription.
+  private _changeLanguageSubscription: Subscription;
+
   /**
    * Constructor.
    *
    * @param _translateService Translate service.
+   * @param _ngxTranslateService NgxTranslateService.
+   */
+  constructor(
+    private _translateService: TranslateService,
+    private _ngxTranslateService: NgxTranslateService
+  ) {}
+
+  /**
+   * Component init hook.
    */
-  constructor(private _translateService: TranslateService) {
-    this._translateService.setLanguage(document.documentElement.lang || 'en');
+  ngOnInit() {
+    this._changeLanguageSubscription = this._ngxTranslateService.onLangChange.subscribe(
+      (event: LangChangeEvent) => {
+        document.cookie = `lang=${event.lang}`;
+      }
+    );
+
+    this._translateService.setLanguage(this._getPreferredLang());
+  }
+
+  /**
+   * Component destruction hook.
+   */
+  ngOnDestroy() {
+    this._changeLanguageSubscription.unsubscribe();
+  }
+
+  /**
+   * Return the preferred language for the user.
+   */
+  private _getPreferredLang(): string {
```

Comment:
Is this can be used to change the frontend language from the backend application?

Reply:
Yes of course. We just have to set a mechanism to read the cookie value in backend application.